optimizely/client/factory.go

```diff
@@ -91,10 +88,10 @@ func (f OptimizelyFactory) ClientWithOptions(clientOptions Options) (*Optimizely
 	} else {
 		// if no context is provided, we create our own cancellable context and hand it over to the client so the client can shut down its child processes
 		ctx = context.Background()
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithCancel(ctx)
-		client.cancelFunc = cancel
 	}
+	var cancel context.CancelFunc
```

Comment:
I don't think this should be moved here because if the user passes in a context, they should take care of making it cancelable. If the testapp needs to be able to cancel it, it should pass in a cancelable context and invoke the cancel func from outside the SDK. @pawels-optimizely does this make sense?